routes/api/v1/meals.js

```diff
@@ -62,4 +71,28 @@ router.delete('/:meal_id/foods/:id', (req, res) => {
     .catch(error => res.status(500).send({error}));
 })
 
+router.get('/most_popular_food', (req, res) => {
+  res.setHeader("Content-Type", "application/json");
+
+  MealFood.findAll({
+    attributes:
+      ['FoodId', [Sequelize.fn('count', Sequelize.col('FoodId')),'count']],
+      group : ['MealFoods.FoodId'],
+      raw: true,
+      order: sequelize.literal('count DESC LIMIT 1')
+  })
+  .then(mostPopular => {
+    console.log(mostPopular)
+    Food.findOne({
+      where: {
+        id: mostPopular[0].FoodId
+      }
+    })
+    .then(food => {
+      res.status(200).send(food)
+    })
+  })
+  .catch(error => res.status(500).send({error}));
+})
```

Comment:
```suggestion
router.get('/most_popular_food', async (req, res) => {
  res.setHeader('Content-Type', 'application/json');
  
  let counted_foods = await MealFood.findAll({
    attributes:
      ['FoodId', [Sequelize.fn('count', Sequelize.col('FoodId')), 'count']],
      group: ['MealFoods.FoodId'],
      raw: true,
      order: sequelize.literal('count DESC LIMIT 1')
  });
  
  res.status(200).send(counted_foods[0]);
});
```

Does this work?

Reply:
This doesn't connect to the Food table for name and calories

Reply:
Heck
```suggestion
router.get('/most_popular_food', async (req, res) => {
  res.setHeader('Content-Type', 'application/json');
  
  let counted_foods = await MealFood.findAll({
    attributes:
    ['FoodId', [Sequelize.fn('count', Sequelize.col('FoodId')), 'count']],
    group: ['MealFoods.FoodId'],
    raw: true,
    order: sequelize.literal('count DESC LIMIT 1')
  });
  let mostPopular = await Food.findOne({where: {id: counted_foods[0].FoodId}});
  res.status(200).send(mostPopular);
});
```
???

Reply:
I tested it and this works. Same thing as before just with async/await instead of promises